plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java

```diff
@@ -447,52 +447,51 @@ protected void provisionFirewallRules(final IpAddress publicIp, final Account ac
         firewallService.applyIngressFwRules(publicIp.getId(), account);
     }
 
+    protected void provisionPublicIpPortForwardingRule(IpAddress publicIp, Network network, Account account,
+                                                       final long vmId, final int sourcePort) throws NetworkRuleConflictException, ResourceUnavailableException {
+        final long publicIpId = publicIp.getId();
+        final long networkId = network.getId();
+        final long accountId = account.getId();
+        final long domainId = account.getDomainId();
+        Nic vmNic = networkModel.getNicInNetwork(vmId, networkId);
+        final Ip vmIp = new Ip(vmNic.getIPv4Address());
+        PortForwardingRuleVO pfRule = Transaction.execute((TransactionCallbackWithException<PortForwardingRuleVO, NetworkRuleConflictException>) status -> {
+            PortForwardingRuleVO newRule =
+                    new PortForwardingRuleVO(null, publicIpId,
+                            sourcePort, sourcePort,
+                            vmIp,
+                            DEFAULT_SSH_PORT, DEFAULT_SSH_PORT,
+                            "tcp", networkId, accountId, domainId, vmId);
+            newRule.setDisplay(true);
+            newRule.setState(FirewallRule.State.Add);
+            newRule = portForwardingRulesDao.persist(newRule);
+            return newRule;
+        });
+        rulesService.applyPortForwardingRules(publicIp.getId(), account);
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.info(String.format("Provisioned SSH port forwarding rule: %s from port %d to 22 on %s to the VM IP : %s in Kubernetes cluster : %s", pfRule.getUuid(), sourcePort, publicIp.getAddress().addr(), vmIp.toString(), kubernetesCluster.getName()));
```

Comment:
nit: we could probably replace the hard coded end port (22) to the constant DEFAULT_SSH_PORT to be safer..

Reply:
Done